pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -161,34 +168,65 @@ func enableSyncerForWorkspace(ctx context.Context, config *rest.Config, workload
 	// Create a service account for the syncer with the necessary permissions. It will
 	// be owned by the workload cluster to ensure cleanup.
 	authResourceName := SyncerAuthResourcePrefix + workloadClusterName
-	sa, err := kubeClient.CoreV1().ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            authResourceName,
-			OwnerReferences: workloadClusterOwnerReferences,
-		},
-	}, metav1.CreateOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to create ServiceAccount: %w", err)
+	sa, err := kubeClient.CoreV1().ServiceAccounts(namespace).Get(ctx, authResourceName,
+		metav1.GetOptions{})
+
+	switch {
+	case errors.IsNotFound(err):
+		if sa, err = kubeClient.CoreV1().ServiceAccounts(namespace).Create(ctx, &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            authResourceName,
+				OwnerReferences: workloadClusterOwnerReferences,
+			},
+		}, metav1.CreateOptions{}); err != nil {
+			return "", fmt.Errorf("failed to create ServiceAccount: %w", err)
+		}
+	case err == nil:
+		//Overwrite the ownerref in case of a user changed it
+		sa.ObjectMeta.OwnerReferences = workloadClusterOwnerReferences
+		if sa, err = kubeClient.CoreV1().ServiceAccounts(namespace).Update(ctx, sa, metav1.UpdateOptions{}); err != nil {
```

Comment:
We try to do `Patch()` instead of `Update()` to avoid potential data loss (old client imports old types that don't know about new fields -> Update will cause the new fields to be dropped, Patch will not). Please look in the repo for examples where we call Patch(). Thanks!